charts/renku/templates/user-pod-egress.yaml

```diff
@@ -0,0 +1,42 @@
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: {{ template "renku.fullname" . }}-user-pod-egress
+  labels:
+    app: {{ template "renku.name" . }}
+    chart: {{ template "renku.chart" . }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  policyTypes:
+  - Egress
+  egress: 
+  - to:
+    # allow DNS resolution (internal and external)
+    ports:
+    - port: 53
+      protocol: UDP
+    - port: 53
+      protocol: TCP
+  - to:
+    # allow connection to internal apps that is necessary
+    - podSelector:
+        matchLabels:
+          app: jupyterhub
```

Comment:
At the moment this allows connection to the jupyterhub proxy and other users notebooks servers too. By specifically selecting the `hub` component here you can narrow this down a little further.